### PR TITLE
AVRO-2127 Make InvalidAvroMagicException thrown if stream header is corrupted

### PR DESCRIPTION
**Pull request by bunadmitrii**
This seems reasonable to throw a subclass of IOException in case if we have an invalid header (empty file for example) so such errors can be handled in user code.

### CODE REVIEW

File: lang/java/avro/src/main/java/org/apache/avro/file/InvalidAvroMagicException.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package org.apache.avro.file;
```
Comment (nandorKollar) [inline lang/java/avro/src/main/java/org/apache/avro/file/InvalidAvroMagicException.java]: Missing license header

**Comment by bunadmitrii** [comment]
Done!

File: lang/java/avro/src/main/java/org/apache/avro/file/DataFileReader.java [hunk 2]
```diff
@@ -47,7 +47,7 @@
                                              DatumReader<D> reader)
     throws IOException {
     if (in.length() < MAGIC.length)
-      throw new IOException("Not an Avro data file");
+      throw new EOFException("Not an Avro data file");
```
Comment (busbey) [inline lang/java/avro/src/main/java/org/apache/avro/file/DataFileReader.java]: Will this be confusing? Say I want to take files and throw them at the wall to see what sticks. With this change as-is I'll need to see if the IOE is either EOF or IAM. Is that desirable? Do we gain enough by being able to distinguish the case where the magic header was invalid due to not enough bytes?

File: lang/java/avro/src/main/java/org/apache/avro/file/DataFileStream.java [hunk 3]
```diff
@@ -102,7 +102,7 @@ void initialize(InputStream in) throws IOException {
       throw new IOException("Not an Avro data file.", e);
```
Comment (busbey) [inline lang/java/avro/src/main/java/org/apache/avro/file/DataFileStream.java]: Is there an analogous case here to the EOF above? If there weren't enough bytes in the stream for the magic check, right now here we throw a bare IOE, right?

**Comment by dkulp** [comment]
This was included as part of #323 / AVRO-2127

**Pull request closed without merging**